runtime-modules/content-directory/src/lib.rs

```diff
@@ -0,0 +1,1645 @@
+        id: PropertyId,
+        access_level: EntityAccessLevel,
+        new_value: PropertyValue<T>,
+        current_prop_value: &mut PropertyValue<T>,
+    ) -> Result<(EntitiesRcVec<T>, EntitiesRcVec<T>), &'static str> {
+        let mut entities_rc_to_increment_vec = vec![];
+        let mut entities_rc_to_decrement_vec = vec![];
+        // Get class-level information about this property
+        if let Some(class_prop) = class.properties.get(id as usize) {
+            // Ensure class property is unlocked for given actor
+            ensure!(
+                !class_prop.is_locked_from(access_level),
+                ERROR_CLASS_PROPERTY_TYPE_IS_LOCKED_FOR_GIVEN_ACTOR
+            );
+
+            // Validate a new property value against the type of this property
+            // and check any additional constraints like the length of a vector
+            // if it's a vector property or the length of a text if it's a text property.
+            class_prop.ensure_property_value_to_update_is_valid(
+                &new_value,
+                entity.get_permissions().get_controller(),
+            )?;
+
+            // Get unique entity ids to update rc
+            if let (Some(entities_rc_to_increment), Some(entities_rc_to_decrement)) = (
+                new_value.get_involved_entities(),
+                current_prop_value.get_involved_entities(),
+            ) {
+                let (entities_rc_to_decrement, entities_rc_to_increment): (
+                    Vec<T::EntityId>,
+                    Vec<T::EntityId>,
+                ) = entities_rc_to_decrement
+                    .into_iter()
+                    .zip(entities_rc_to_increment.into_iter())
+                    .filter(|(entity_rc_to_decrement, entity_rc_to_increment)| {
+                        entity_rc_to_decrement != entity_rc_to_increment
+                    })
+                    .unzip();
+                entities_rc_to_increment_vec.push(entities_rc_to_increment);
+                entities_rc_to_decrement_vec.push(entities_rc_to_decrement);
+            }
+
+            // Update a current prop value in a mutable vector, if a new value is valid.
+            current_prop_value.update(new_value);
+        }
+        Ok((entities_rc_to_increment_vec, entities_rc_to_decrement_vec))
+    }
+
+    fn get_property_value_vec(
+        entity: &Entity<T>,
+        in_class_schema_property_id: PropertyId,
+    ) -> Result<&VecPropertyValue<T>, &'static str> {
+        entity
+            .values
+            .get(&in_class_schema_property_id)
+            // Throw an error if a property was not found on entity
+            // by an in-class index of a property.
+            .ok_or(ERROR_UNKNOWN_ENTITY_PROP_ID)?
+            .as_vec_property_value()
+            // Ensure prop value under given class schema property id is vector
+            .ok_or(ERROR_PROP_VALUE_UNDER_GIVEN_INDEX_IS_NOT_A_VECTOR)
+    }
+
+    fn get_entity_and_access_level(
```

Comment:
This is an ensure.

Reply:
Done https://github.com/Joystream/joystream/pull/470/commits/ac6e00a1bdf1bb3125e8764984f1256a94bc85d8